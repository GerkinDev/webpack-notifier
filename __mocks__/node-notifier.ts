export const notify = jest.fn(() => void 0);
