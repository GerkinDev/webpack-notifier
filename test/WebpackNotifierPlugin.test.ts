import {notify} from 'node-notifier';
import WebpackNotifierPlugin from '../';
import {getCompiler, compile, prepareFs, contentImageSerializer, reduceArraySerializer} from './utils';
import fixtures from "./fixtures";

expect.addSnapshotSerializer(reduceArraySerializer);
expect.addSnapshotSerializer(contentImageSerializer);

describe('WebpackNotifierPlugin', () => {
  describe('one compilation', () => {
    test.each([
      [['successful'], {}],
      [['error'], {}],
      [['warning'], {}],
    ])('%j %j', testChangesFlow);
  });
});

async function testChangesFlow(sources, opts)  {
  const compiler = getCompiler();
  const plugin = new WebpackNotifierPlugin(opts);
  plugin.apply(compiler);

  for (const name of sources) {
    notify.mockClear();
    prepareFs(fixtures.simple[name]);
    await compile(compiler);
    expect(notify.mock.calls).toMatchSnapshot();
  }
}
