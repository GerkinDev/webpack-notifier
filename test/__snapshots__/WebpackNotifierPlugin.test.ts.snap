// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebpackNotifierPlugin one compilation ["error"] {} 1`] = `
Object {
  "contentImage": "__dirname/logo.png",
  "message": "Error: /entry.js
SyntaxError: Unexpected token (1:8)",
  "title": undefined,
}
`;

exports[`WebpackNotifierPlugin one compilation ["successful"] {} 1`] = `
Object {
  "contentImage": "__dirname/logo.png",
  "message": "Build successful",
  "title": undefined,
}
`;

exports[`WebpackNotifierPlugin one compilation ["warning"] {} 1`] = `
Object {
  "contentImage": "__dirname/logo.png",
  "message": "Warning: /entry.js
require.extensions is not supported by webpack. Use a loader instead.",
  "title": undefined,
}
`;
